components/logins/src/login.rs

```diff
@@ -84,7 +331,46 @@ impl Login {
         self.guid.as_str()
     }
 
+    // Checks whether the Login is valid, without attempting to fix any fields.
+    // Returns an error if invalid data is found, even if it could have been fixed.
     pub fn check_valid(&self) -> Result<()> {
+        self.validate_and_fixup(false)?;
+        Ok(())
+    }
+
+    // Return either the existing login, a fixed-up verion, or an error.
```

Comment:
Nit: `///` is for item-level doc comments in rust. Here and elsewhere (except for the module-level comment, which is correctly `//!`).